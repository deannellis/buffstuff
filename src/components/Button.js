import React from 'react';

const Button = (props) => {
    return (
        <button 
            className={`button ${props.variant ? 'button--' + props.variant : ''} ${props.disabled ? 'button--disabled' : ''}`}
            type={props.type}
        >
            {props.children}
        </button>
    );
}

Button.defaultProps = {
    disabled: false,
    type: 'button'
}
 
export default Button;