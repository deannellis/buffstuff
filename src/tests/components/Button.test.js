import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../components/Button';

test('should render Button without props', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
});

test('should render Button with props', () => {
    const wrapper = shallow(<Button variant="primary" disabled="true" type="submit" >Button</Button>);
    expect(wrapper).toMatchSnapshot();
});