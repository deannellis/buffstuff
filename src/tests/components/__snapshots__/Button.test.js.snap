// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button with props 1`] = `
<button
  className="button button--primary button--disabled"
  type="submit"
>
  Button
</button>
`;

exports[`should render Button without props 1`] = `
<button
  className="button  "
  type="button"
/>
`;
